src/stdlib_bitsets_64.fypp

```diff
@@ -0,0 +1,1120 @@
+        character(*), parameter :: procedure = "OUTPUT"
+
+        write(unit, iostat=ierr, iomsg=message) self % num_bits
+        if (ierr /= 0) go to 999
+
+        if (self % num_bits < 1) return
+        write(unit, iostat=ierr, iomsg=message) self % block
+        if (ierr /= 0) go to 999
+
+        return
+
+999     call error_handler( 'Failure on a WRITE statement for UNIT.', &
+                            write_failure, status, module_name, procedure )
+
+    end subroutine output_64
+
+
+    module subroutine read_bitset_string_64(self, string, status)
+!
+!     Uses the bitset literal in the default character `string`, to define
+!     the bitset, `self`. The literal may be preceded by an an arbitrary
+!     sequence of blank characters. If `status` is absent an error results
+!     in an error stop with an informative stop code. If `status`
+!     is present it has one of the values
+!     * `success` - if no problems occurred,
+!     * `alloc_fault` - if allocation of memory for SELF failed,
+!     * `array_size_invalid_error - if `bits(self)` in `string` is greater
+!       than 64 for a `bitset_64`,
+!     * `char_string_invalid_error` - if the bitset literal has an invalid
+!       character,
+!     * `char_string_too_small_error - if the string ends before all the bits
+!       are read.
+!     * `integer_overflow_error` - if the bitset literal has a `bits(self)`
+!       value too large to be represented,
+!
+        class(bitset_64), intent(out)  :: self
+        character(len=*), intent(in)   :: string
+        integer, intent(out), optional :: status
+
+        integer(bits_kind)      :: bit, bits
+        integer(bits_kind)      :: digits, pos
+        character(*), parameter :: procedure = "READ_BITSET"
+        integer                 :: stat
+
+        pos = 1
+        find_start: do pos=1, len(string)
+            if ( string(pos:pos) /= ' ' ) exit
+        end do find_start
+
+        if ( pos > len(string) - 8 ) go to 999
+
+        if ( string(pos:pos) /= 's' .AND. string(pos:pos) /= 'S' ) go to 999
+
+        pos = pos + 1
+        bits = 0
+        digits = 0
+
+        do
+            select case( iachar( string(pos:pos) ) )
+            case(ia0:ia9)
+                digits = digits + 1
+                if ( digits == 10 .AND. bits > 2_bits_kind**30/5 ) go to 996
+!! May not be quite right
+                if ( digits > 10 ) go to 996
+                bits = bits*10 + iachar( string(pos:pos) ) - ia0
+                if ( bits < 0 ) go to 996
+            case(iachar('b'), iachar('B'))
+                go to 100
+            case default
+                go to 999
+            end select
+
+            pos = pos + 1
+
+        end do
+
+100     if ( bits > 64 ) then
+            call error_handler( 'BITS in STRING was greater than 64.', &
+                                char_string_too_large_error, status,   &
+                                module_name, procedure )
+            return
+        end if
```

Comment:
I recently introduced `max_digits` and `overflow_bits` in stdlib_bitsets.fypp
      integer, parameter ::                   &
          max_digits = 10 ! bits_kind == int32
    !        max_digits = 20 ! bits_kind == int64

      integer(bits_kind), parameter ::     &
          overflow_bits = 2_bits_kind**30/5 ! bits_kind == int32
    !        overflow_bits = 2_bits_kind**62/5 ! bits_kind == int64

so that

                if (  digits == 10 .AND. bits > 2_bits_kind**30/5 ) &

can be written

                if (  digits == max_digits .AND. bits > overflow_bits ) &

so that overflow checks for reading and writing bitsets can be easily modified for bits_kind=int64. For some reason, this introduction has revealed that gfortran 10.2's runtime code generation is fragile. In order to get the test cases to run to completion I had to introduce the statement

    public :: max_digits, overflow_bits

in stdlib_bitsets.fypp and change

        if ( pos < 0 .OR. pos > self % num_bits-1 ) return
        self % block = ibset( self % block, pos )

in stdlib_bitsets_64.fypp to

        integer(block_kind) :: dummy
        if ( pos < 0 .OR. pos > self % num_bits-1 ) return
        dummy = ibset( self % block, pos )
        self % block = dummy

in order for the test cases to run to completion. I don't think that either change is required by the standard. Certainly other parameters used by the submodules do not have to be `public`, and the comparable code in stdlib_bitsets_large.fypp

        if ( pos < 0 .OR. pos > self % num_bits-1 ) return

        set_block = pos / block_size + 1
        block_bit = pos - (set_block - 1) * block_size
        self % blocks(set_block) = ibset( self % blocks(set_block), block_bit )

still seems to be working.